pkg/scheduler/framework/plugins/nodevolumelimits/csi_test.go

```diff
@@ -305,6 +306,17 @@ func TestCSILimits(t *testing.T) {
 			test:             "should count in-tree volumes if migration is enabled",
 			wantStatus:       framework.NewStatus(framework.Unschedulable, ErrReasonMaxVolumeCountExceeded),
 		},
+		{
+			newPod:           inTreeInlineVolPod,
+			existingPods:     []*v1.Pod{inTreeTwoVolPod},
+			filterName:       "csi",
+			maxVols:          2,
+			driverNames:      []string{csilibplugins.AWSEBSInTreePluginName, ebsCSIDriverName},
+			migrationEnabled: true,
+			isNilCsiNode:     true,
+			limitSource:      "csinode",
```

Comment:
are you able to repro the error if you set `limitSource` to `node` and without needing `isNilCsiNode`?

Reply:
We need the isNilCsiNode to generate the nil pointer exception? If I commented out the csiNodeName and log the csiNode.Name the test returns error:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x13ba74d]
```

Reply:
when `limitSource == node`, I believe that should also cause csiNode to be nil: https://github.com/kubernetes/kubernetes/blob/250497ebca681fa8c054cac45bb5d8438b67d7a4/pkg/scheduler/framework/plugins/nodevolumelimits/csi_test.go#L663

Reply:
I tried again the same test case as the comment above:

```
{
    newPod:           inTreeOneVolPod,
    existingPods:     []*v1.Pod{inTreeTwoVolPod},
    filterName:       "csi",
    maxVols:          2,
    driverNames:      []string{csilibplugins.AWSEBSInTreePluginName, ebsCSIDriverName},
    migrationEnabled: true,
    limitSource:      "node",
    test:             "should count in-tree volumes if migration is enabled",
    wantStatus:       framework.NewStatus(framework.Unschedulable, ErrReasonMaxVolumeCountExceeded),
}
```

It did not trigger the csiNode == nil, it did not go into the ```if !isCSIMigrationOn(csiNode, inTreeProvisionerName)``` code path.

Reply:
I checked out your PR, reverted csi.go, and tweaked the test inputs and was able to repro the crash:
```
--- a/pkg/scheduler/framework/plugins/nodevolumelimits/csi_test.go
+++ b/pkg/scheduler/framework/plugins/nodevolumelimits/csi_test.go
@@ -313,8 +313,8 @@ func TestCSILimits(t *testing.T) {
                        maxVols:          2,
                        driverNames:      []string{csilibplugins.AWSEBSInTreePluginName, ebsCSIDriverName},
                        migrationEnabled: true,
-                       isNilCsiNode:     true,
-                       limitSource:      "csinode",
+                       isNilCsiNode:     false,
+                       limitSource:      "node",
                        test:             "nil csi node",
                },
```

Reply:
> We need the isNilCsiNode to generate the nil pointer exception? I commented out the csiNodeName and log the csiNode.Name the test returns error:
> 
> ```
> panic: runtime error: invalid memory address or nil pointer dereference [recovered]
> 	panic: runtime error: invalid memory address or nil pointer dereference
> [signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x13ba74d]
> ```

This was ran with:

```
{
			newPod:           inTreeInlineVolPod,
			existingPods:     []*v1.Pod{inTreeTwoVolPod},
			filterName:       "csi",
			maxVols:          2,
			driverNames:      []string{csilibplugins.AWSEBSInTreePluginName, ebsCSIDriverName},
			migrationEnabled: true,
			isNilCsiNode:     true,
			limitSource:      "csinode",
}
```

without the change:

```
csiNodeName := ""
if csiNode != nil {
	csiNodeName = csiNode.Name
}
```

also generated the nil pointer exception, do we need another test case specific to
```
 isNilCsiNode:     false,
limitSource:      "node",
```?

Reply:
correct

Reply:
Updated